pytorch_lightning/core/step_result.py

```diff
@@ -0,0 +1,327 @@
+from typing import Optional, Dict, Union
+from torch import Tensor
+import torch
+from copy import copy
+
+
+class Result(Dict):
```

Comment:
Not sure if we really should derive this from `dict` or from `collections.MutableMapping`. Either way, if we derive from `dict` it must be the type `dict` and not the type annotation `Dict`

Reply:
probably `collections.MutableMapping` is better solution

Reply:
I could not agree more. We should definitely not inherit from dict, it causes too many issues. For example, update, pop etc. stop working unless you completely re-implement them.

Reply:
just tested and in such case we need to define own methods:
__delitem__, __getitem__, __iter__, __len__, __setitem__

Reply:
ok, once we merge these PRs we can change to subclass from this